src/services/study.service.ts

```diff
@@ -56,6 +57,56 @@ function createEmptyStudySession(
   return studySession
 }
 
+async function editStudyLogo(
+  studyId: string,
+  token: string,
+  fileSize: number,
+  fileName: string,
+  fileBody: string,
+  mimeType: string,
+  file: File,
+) {
+  try {
+    // upload to the backend
+    const uploadToBackendEndpoint = `${constants.endpoints.studies}/${studyId}/logo`
+    const bodyForUploadBackendEndpoint = {
+      name: fileName,
+      mimeType: mimeType,
+      size: fileSize,
+      fileGuid: fileBody,
+    } as FileRevision
+    const uploadResult = await callEndpoint(
+      uploadToBackendEndpoint,
+      'POST',
+      bodyForUploadBackendEndpoint,
+      token,
+    )
+    const uploadData = uploadResult.data as FileRevision
+    // upload to aws
+    let headers: HeadersInit = new Headers()
+    headers.set('Content-Type', mimeType)
+    headers.set('Content-Disposition', 'inline')
+    const config = {
+      method: 'PUT',
+      headers,
+      body: file,
+    }
+    const uploadURL = uploadData.uploadURL!
+    await fetch(uploadURL, config)
+    // Tell the backend that we have successfully finished uploading file to aws
+    const successfulUploadEndpoint = `${constants.endpoints.studies}/${studyId}/logo/${uploadData.createdOn}`
+    const successulyUploadResponse = await callEndpoint(
```

Comment:
callEndpoint is a generic function (https://www.typescriptlang.org/docs/handbook/2/generics.html)  so it is good practice to specify the type callEndpoint<Study> in this case